[P008] Allow selection of 26..64 bits receive format

tonhuisman opened the pull request:
Proposed solution in PR #2400 is not effective (data pins could also be swapped), the plugin has been updated to allow 26..64 bits data format receiving, resulting in effectively 24..62 bits of data. (To allow for future extension.)
Comments in #2400 show receiving 35 bits of data, from (at least) a 37 bit data stream.
- Extends selection of Wiegand Type from 26/34 bits to 26..64 bits
- Minor code optimizations

TODO:
- [ ] Testing (help wanted, please)
- [x] Documentation update

Resolves #414
(NB: I didn't want to hijack PR #2400, but this PR is supposed to replace it)

tonhuisman committed:
[P008] Add 'Alternative decoding' option

tonhuisman committed:
[P008] Format code using Uncrustify

TD-er commented:
I justy read through both the PR and issue you linked and I'm not sure this is "the" fix.

tonhuisman commented:
It should effectively be the same fix as presented in the PR, swapping the interrupt handlers is IMHO the same as connecting them reversed. Though I can't oppose the fact that it may not be "the" fix, as I have no Wiegand hardware to confirm.

tonhuisman committed:
[P008] Format code minor improvement

TD-er commented:
Shardan mentioned a few times it wasn't a fix for his tests.
I do agree your changes reflect the original PR's intentions. But I'm afraid it may be later something like a combobox with very cryptic selections like:
- Alternative decoding
- Alternative2 for Wiegand 34 bit
- Yet another alternative, mixing alt & alt2
- Alternative for 35-three-quarter bit cards
- Patch for southern hemisphere cards scanned left handed
- etc.

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p008-add-option-for-reversed-decoding

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p008-add-option-for-reversed-decoding

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p008-add-option-for-reversed-decoding

tonhuisman committed:
Merge branch 'mega' of https://github.com/letscontrolit/ESPEasy into feature/p008-add-option-for-reversed-decoding

tonhuisman committed:
[P008] Reverted previous mod. Changed to allow 26..64 bits receive format. Updated documentation

tonhuisman commented:
I have reverted the previous modification, and implemented support for receiving 26..64 bits via Wiegand protocol (don't know if that is actually supported, but this plugin can now handle it).

@shardanX are you available to test with the 35 bit data (probably 37 bits received) format you reported as not working correctly? If needed a .bin download can be made available.

tonhuisman committed:
Merge branch 'mega' into feature/p008-add-option-for-reversed-decoding

tonhuisman committed:
Merge branch 'mega' into feature/p008-add-option-for-reversed-decoding

tonhuisman commented:
😄
